packages/amplify-e2e-core/src/utils/rds.ts

```diff
@@ -273,14 +278,14 @@ export const setupDataInExistingCluster = async (identifier: string, config: RDS
     }
 
     const clusterArn = dbClusterObj.DBClusterArn;
-    const secretArn = dbClusterObj.MasterUserSecret.SecretArn;
+    const masterSecretArn = dbClusterObj?.MasterUserSecret?.SecretArn || secretArn;
```

Comment:
Typically you'd use `secretArn` if provided, and fall back to the master only if it wasn't present. As written, callers have no way to use users with a reduced scope if a Master User secret exists in the cluster config. 

Either way, you've no guarantee that the resolved value is a master secret, right? If you end up using the incoming argument, all you know is that it's a secret, not necessarily a DB master secret.